samples/net/sockets/echo_client/README.rst

```diff
@@ -0,0 +1,63 @@
+.. _sockets-echo-client-sample:
+
+Echo Client
+###########
+
+Overview
+********
+
+The echo-client sample application for Zephyr implements a UDP/TCP client
+that will send IPv4 or IPv6 packets, wait for the data to be sent back,
+and then verify it matches the data that was sent.
+
+The source code for this sample application can be found at:
+:file:`samples/net/sockets/echo_client`.
+
+Requirements
+************
+
+- :ref:`networking_with_qemu`
+
+Building and Running
+********************
+
+There are multiple ways to use this application. One of the most common
+usage scenario is to run echo-client application inside QEMU. This is
+described in :ref:`networking_with_qemu`.
+
+Build echo-client sample application like this:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/sockets/echo_client
+   :board: <board_to_use>
+   :goals: build
+   :compact:
+
+``board_to_use`` defaults to ``qemu_x86``. In this case, you can run the
+application in QEMU using ``make run``. If you used another BOARD, you
+will need to consult its documentation for application deployment
+instructions. You can read about Zephyr support for specific boards in
+the documentation at :ref:`boards`.
+
+Running echo-server in Linux Host
+=================================
+
+There is one useful testing scenario that can be used with Linux host.
+Here echo-client is run in QEMU and echo-server is run in Linux host.
+
```

Comment:
It's worth adding a note you have in the echo-server documentation here too (but with the server and client running in the opposite environments):

    Note that echo-server must be running in the Linux host terminal window
    before you start the echo-client application in QEMU.

Reply:
Done.

Reply:
Yeah, I see that the docs are mostly copy-paste from the native API client/server samples, and I don't see them mentioning net-tools at all (so, it's not very clear how to "run something on Linux". But that needs to be fixed in those samples first/too.